src/snowflake/snowpark/column.py

```diff
@@ -934,21 +943,41 @@ def _cast(
             )
             ast.col.CopyFrom(self._ast)
             to._fill_ast(ast.to)
-        return Column(Cast(self._expression, to, try_), _ast=expr, _emit_ast=_emit_ast)
+        return Column(
+            Cast(self._expression, to, try_, is_rename, is_add),
+            _ast=expr,
+            _emit_ast=_emit_ast,
+        )
 
     @publicapi
-    def cast(self, to: Union[str, DataType], _emit_ast: bool = True) -> "Column":
+    def cast(
+        self,
+        to: Union[str, DataType],
+        _emit_ast: bool = True,
+        is_rename: bool = False,
+        is_add: bool = False,
```

Comment:
I feel those two should put in front of the private parameter `_emit_ast`

Reply:
+1